computation.hpp

```diff
@@ -266,15 +266,23 @@ double computePlanningUnitValue(const spustuff &pu,const sconnections &connectio
     return(theValue);
 }
 
+
 // Returns both index and species amount for a planning unit
 inline
 pair<int,double> returnAmountSpecAtPu(const spustuff &pu, const vector<spu> &SM, int iSpecIndex)
 {
     if (pu.richness > 0)
-        for (int i=0;i<pu.richness;i++)
-            if (SM[pu.offset + i].spindex == iSpecIndex)
-                return pair<int,double>(pu.offset + i, SM[pu.offset + i].amount);
-
+    {

```

Comment:
Out of curiosity what was the rough speedup gain here when you ran this on a bigger dataset?